rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -162,14 +169,36 @@ signal_handler(int signal_value)
 }
 
 void
-rclcpp::init(int argc, char const * const argv[])
+rclcpp::init(int argc, char const * const argv[], const rclcpp::InitOptions & init_options)
 {
-  g_is_interrupted.store(false);
-  if (rcl_init(argc, argv, rcl_get_default_allocator()) != RCL_RET_OK) {
-    std::string msg = "failed to initialize rmw implementation: ";
-    msg += rcl_get_error_string().str;
-    rcl_reset_error();
-    throw std::runtime_error(msg);
+  using rclcpp::contexts::default_context::get_global_default_context;
+  init_local(argc, argv, get_global_default_context(), init_options);
+  // Install the signal handlers.
+  rclcpp::install_signal_handlers();
+}
+
+void
+rclcpp::init_local(
+  int argc,
+  char const * const argv[],
+  rclcpp::Context::SharedPtr context_out,
+  const rclcpp::InitOptions & init_options)
+{
+  auto rcl_context_wrapper = context_out->get_sub_context<RclContextWrapper>();
+  rcl_ret_t ret = rcl_init(
+    argc, argv, init_options.get_rcl_init_options(), rcl_context_wrapper->get_context().get());
+  if (RCL_RET_OK != ret) {
+    rclcpp::exceptions::throw_from_rcl_error(ret, "failed to initialize rcl: ");
```

Comment:
nit: other calls to this function don't append `: `
```suggestion
    rclcpp::exceptions::throw_from_rcl_error(ret, "failed to initialize rcl");
```